walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -108,4 +109,83 @@
     Collections.sort(timeConflicts, TimeRange.ORDER_BY_START);
     return timeConflicts;
   }
+
+  /**
+   * Find the optimal time ranges for optional attendees.
+   *
+   * @param events
+   * @param request
+   * @return Returns the optimal time ranges for optional attendees. Can be empty.
+   */
+  private Collection<TimeRange> findOptimalOptionalAttendee(
+      Collection<Event> events, MeetingRequest request) {
+    // This works by checking each combination of optional attendees in increasing quantity.
+    // Starting with groups of one attendee, then two, etc. Until less than two combinations
+    // had non-empty ranges (see optimization notes below) or all combinations have been checked.
+    // 
+    // For large numbers of optional attendees, this can quickly become inefficient.
+    // There is room here for a dynamic programming approach following the logic that a combination
+    // of three elements could be more efficiently computed by using a previously-computed
+    // combination of two elements. 
+
+    ArrayList<String> optionalAttendees = 
+        new ArrayList<String>(request.getOptionalAttendees());
+    Collection<String> attendees = request.getAttendees();
+
+    int successfulCombinations = 0;
+    Collection<TimeRange> bestTimeRange = Arrays.asList();
+    int longestDuration = 0;
```

Comment:
just declare this at the start of the for loop. That way you won't have to have the extra line resetting this to 0 at the end of the for loop.

Reply:
This is also awkward. That's what you get by staring at the same block of code for too long. Thanks for the catch.